lib/net/ldap.rb

```diff
@@ -553,6 +555,10 @@ def initialize(args = {})
     @force_no_page = args[:force_no_page] || DefaultForceNoPage
     @encryption = normalize_encryption(args[:encryption]) # may be nil
     @connect_timeout = args[:connect_timeout]
+    if args[:connection_pool].is_a? Hash
+      options = { size: 5, timeout: (@connection_timeout || 5) }.merge args[:connection_pool]
```

Comment:
I'm not sure if we should default the timeout here in this way. The time to wait for a connection to free up is not quite the same as waiting for a single connection to be made. Perhaps we should default so some higher value here like 30 seconds.